compositor_render/src/transformations/shader/validation.rs

```diff
@@ -556,6 +563,95 @@ impl TypeInnerExt for naga::TypeInner {
         }
     }
+    fn to_string(&self, module: &naga::Module) -> String {
+        match self {
+            naga::TypeInner::Scalar { kind, width } => kind.to_string(*width),
+            naga::TypeInner::Vector { size, kind, width } => {
+                format!("vec{}<{}>", *size as u8, kind.to_string(*width))
+            }
+            naga::TypeInner::Matrix { .. } => "matrix".to_string(),
+            naga::TypeInner::Atomic { .. } => "atomic".to_string(),
+            naga::TypeInner::Pointer { .. } => "pointer".to_string(),
+            naga::TypeInner::ValuePointer { .. } => "value pointer".to_string(),
+            naga::TypeInner::Array { .. } => "array".to_string(),
+            naga::TypeInner::Struct { .. } => "struct".to_string(),
+            naga::TypeInner::Image {
+                dim,
+                arrayed,
+                class,
+            } => match (dim, arrayed, class) {
+                (naga::ImageDimension::D1, false, naga::ImageClass::Sampled { kind, .. }) => {
+                    format!("texture_1d<{}>", kind.to_string(4))
+                }
+                (naga::ImageDimension::D2, false, naga::ImageClass::Sampled { kind, .. }) => {
+                    format!("texture_2d<{}>", kind.to_string(4))
+                }
+                (naga::ImageDimension::D2, true, naga::ImageClass::Sampled { kind, .. }) => {
+                    format!("texture_2d_array<{}>", kind.to_string(4))
+                }
+                (naga::ImageDimension::D3, false, naga::ImageClass::Sampled { kind, .. }) => {
+                    format!("texture_3d<{}>", kind.to_string(4))
+                }
+                (naga::ImageDimension::Cube, false, naga::ImageClass::Sampled { kind, .. }) => {
+                    format!("texture_cube<{}>", kind.to_string(4))
+                }
+                (naga::ImageDimension::Cube, true, naga::ImageClass::Sampled { kind, .. }) => {
+                    format!("texture_cube_array<{}>", kind.to_string(4))
+                }
+                _ => format!("{:?}", self),
+            },
+            naga::TypeInner::Sampler { .. } => "sampler".to_string(),
+            naga::TypeInner::AccelerationStructure => "acceleration structure".to_string(),
+            naga::TypeInner::RayQuery => "ray query".to_string(),
+            naga::TypeInner::BindingArray { base, size } => {
+                let size: Option<&naga::Constant> = match size {
+                    ArraySize::Constant(size) => Some(&module.constants[*size]),
+                    ArraySize::Dynamic => None, // TODO: not sure how to handle this
+                };
+                let base: &naga::Type = &module.types[*base];
+                format!(
+                    "binding_array<{}, {}>",
+                    base.inner.to_string(module),
+                    size.map(|t| t.inner.to_string(module))
+                        .unwrap_or("_".to_string())
```

Comment:
unwrap will happen when BindingArray would have dynamic size which should not happen as far as I understand